app/src/main/java/de/danoeh/antennapod/fragment/actions/FeedMultiSelectActionHandler.java

```diff
@@ -0,0 +1,189 @@
+    }
+
+    public void handleAction(int id) {
+        if (id == R.id.remove_item) {
+            deleteChecked();
+        } else if (id == R.id.keep_updated) {
+            keepUpdatedChecked();
+        } else if (id == R.id.autodownload) {
+            autoDownloadChecked();
+        } else if (id == R.id.autoDeleteDownload) {
+            autoDeleteEpisodesChecked();
+        } else if (id == R.id.playback_speed) {
+            playbackSpeedChecked();
+        } else {
+            Log.e(TAG, "Unrecognized speed dial action item. Do nothing. id=" + id);
+        }
+    }
+
+    private void autoDownloadChecked() {
+        PreferenceSwitchDialog preferenceSwitchDialog = new PreferenceSwitchDialog(activity,activity.getString(R.string.auto_download_settings_label), activity.getString(R.string.auto_download_label));
+        preferenceSwitchDialog.setOnPreferenceChangedListener(new PreferenceSwitchDialog.OnPreferenceChangedListener() {
+            @Override
+            public void preferenceChanged(boolean enabled) {
+                    saveFeedPreferences(feedPreferences -> feedPreferences.setAutoDownload(enabled));
+                    ApplyToEpisodesDialog dialog = new ApplyToEpisodesDialog(activity, enabled);
+                    dialog.createNewDialog().show();
+            }
+        });
+        preferenceSwitchDialog.openDialog();
+    }
+
+    public static final float SPEED_USE_GLOBAL = -1;
+    private static final DecimalFormat SPEED_FORMAT =
+            new DecimalFormat("0.00", DecimalFormatSymbols.getInstance(Locale.US));
+
+    private void playbackSpeedChecked() {
+        final String[] speeds = activity.getResources().getStringArray(R.array.playback_speed_values);
+        String[] values = new String[speeds.length + 1];
+        values[0] = SPEED_FORMAT.format(SPEED_USE_GLOBAL);
+
+        String[] entries = new String[speeds.length + 1];
+        entries[0] = activity.getString(R.string.feed_auto_download_global);
+
+        System.arraycopy(speeds, 0, values, 1, speeds.length);
+        System.arraycopy(speeds, 0, entries, 1, speeds.length);
+
+        PreferenceListDialog preferenceListDialog = new PreferenceListDialog(activity, activity.getString(R.string.playback_speed));
+        preferenceListDialog.openDialog(entries);
+        preferenceListDialog.setOnPreferenceChangedListener(pos -> {
+            saveFeedPreferences(feedPreferences -> {
+                feedPreferences.setFeedPlaybackSpeed(Float.parseFloat((String) values[pos]));
+                // TODO: 8/5/2021  
+//                EventBus.getDefault().post(
+//                        new SpeedPresetChangedEvent(feedPreferences.getFeedPlaybackSpeed(), feed.getId()));
+            });
+            
+        });
+    }
+
+    private void autoDeleteEpisodesChecked() {
+        PreferenceListDialog preferenceListDialog = new PreferenceListDialog(activity, "Auto delete episodes");
+        String[] items = new String[]{"global", "always", "never"};
```

Comment:
This works only in English. Please use `@array/spnAutoDeleteItems` and `@array/spnAutoDeleteValues`